### PR TITLE
Add http scheme if no scheme is provided when creating a HttpRequest …

### PR DESCRIPTION
**Pull request by daschult**
…in RestProxy

### CODE REVIEW

File: client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import rx.Single;
 import rx.exceptions.Exceptions;
 import rx.functions.Func1;
+import sun.plugin.dom.exception.InvalidStateException;
```
Comment (RikkiGibson) [inline client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: Checkstyle might be mad about this?

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: I always forget to run checkstyle after I make little changes to a pull request. I'll fix this.

File: client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java [hunk 2]
```diff
@@ -220,7 +220,7 @@ public String httpMethod() {
     public String scheme(Object[] swaggerMethodArguments) {
         final String substitutedHost = applySubstitutions(rawHost, hostSubstitutions, swaggerMethodArguments, UrlEscapers.urlPathSegmentEscaper());
         final String[] substitutedHostParts = substitutedHost.split("://");
-        return substitutedHostParts.length < 1 ? "https" : substitutedHostParts[0];
+        return substitutedHostParts.length < 1 ? null : substitutedHostParts[0];
```
Comment (RikkiGibson) [inline client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java]: Is the plan now that we're not going to pick a default scheme for users? As in, if Swagger didn't provide it and the host parameter value didn't provide it, then we just pass it on to the HTTP client and it does whatever it decides to do?

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java]: Yes. We have some tests in autorest.java that will run into this very scenario, which is why I created the ProtocolPolicy RequestPolicy. If no protocol is provided and the HttpRequest hits the HttpClient, then the HttpClient will most likely throw an exception.

**Comment by RikkiGibson** [comment]
Sounds good.

**Pull request merged**